url529c09a30c_kianyangchn_DriveNow_tgz-pJ8-drivenow_tests_TestJ8/src/drivenow/controls/Ressourcenverwaltung.java

```diff
@@ -0,0 +1,326 @@
+package drivenow.controls;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+import drivenow.*;
+import drivenow.Systemereignis.Nachricht;
+import drivenow.entities.*;;
+
+public class Ressourcenverwaltung {
+
+	private static Ressourcenverwaltung instance;
+	private ArrayList<Auto> autos;
+	private ArrayList<Fahrlehrerindaten> fahrlehrerin;
+	private ArrayList<Fahrschueler> fahrschueler;
+	
+	public Ressourcenverwaltung() {
+		this.autos = new ArrayList<Auto>();
+		this.fahrlehrerin = new ArrayList<Fahrlehrerindaten>();
+		this.fahrschueler = new ArrayList<Fahrschueler>();
+	}
+	
+	public static Ressourcenverwaltung getInstance() {
+		if (instance == null) {
+			instance = new Ressourcenverwaltung();
+		}
+		return instance;
+	}
+	
+	public void cleanInstance() {
+		instance=null;
+	}
+	
+	/**
+	 * Geben alle Autos zurueck
+	 * @return
+	 * 		Die gesamte Automenge.
+	 */
+	public ArrayList<Auto> get_auto() {
+		return this.autos;
+	}
+	
+	/**
+	 * Geben alle Fahrlehrerinen zurueck.
+	 * @return
+	 * 		Die gesamte Fahrlehrerinmenge.
+	 */
+	public ArrayList<Fahrlehrerindaten> get_fahrlehrerin() {
+		return this.fahrlehrerin;
+	}
+	
+	/**
+	 * Geben alle Fahrschueleren zurueck.
+	 * @return
+	 * 		Die gesamte Fahrschuelermenge.
+	 */
+	public ArrayList<Fahrschueler> get_fahrschueler() {
+		return this.fahrschueler;
+	}
+	
+	/**
+	 * Geben ein Auto mit dem gegebene Kennzeichen zurueck. 
+	 * @param kennzeichen
+	 * 		Kennzeichen eines Autos.
+	 * @return
+	 * 		Ein Auto mit dem entsprechende Kennzeichen.
+	 */
+	private Auto find_auto(String kennzeichen) {
+		if (this.autos.size()!=0) {
+			for (int i=0;i<this.autos.size();i++) {
+				if (this.autos.get(i).get_kennzeichen()==kennzeichen) {
+					return this.autos.get(i);
+				}
+			}
+		}
+		return null;
+	}
+	
+	/**
+	 * Geben ein Fahrlehrerin mit dem gegebene Name zurueck.
+	 * @param name
+	 * 		Name eines Fahrlehrerin.
+	 * @return
+	 * 		Ein Fahrlehrerin mit dem entsprechende Name. 
+	 */
+	private Fahrlehrerindaten find_fahrlehrerin(String name) {
+		if (this.fahrlehrerin.size()!=0) {
+			for (int i=0;i<this.fahrlehrerin.size();i++) {
+				if (this.fahrlehrerin.get(i).get_name()==name) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.